1.2/sd-client/src/main/java/ServiceDirectoryClient.java

```diff
@@ -0,0 +1,256 @@
+     */
+    private static Map<CMD, String> commands = new LinkedHashMap<>();
+
+    enum CMD {
+        getAllServices(0),
+        getInstanceOf(1),
+        getAllInstancesOf(1),
+        testConnection(0);
+        private final int sizeOfArgs;
+
+        CMD(int sizeOfArgs) {
+            this.sizeOfArgs = sizeOfArgs;
+        }
+
+        public int getSizeOfArgs() {
+            return sizeOfArgs;
+        }
+    }
+
+    /**
+     * The static initializer for commands
+     */
+    static {
+        commands.put(CMD.getAllServices, "");
+        commands.put(CMD.getInstanceOf, "<serviceName>");
+        commands.put(CMD.getAllInstancesOf, "<serviceName>");
+        commands.put(CMD.testConnection, "");
+    }
+
+    /**
+     * Main function.
+     *
+     * @param args arguments.
+     */
+    public static void main(String[] args) {
+        ServiceDirectoryClient client = new ServiceDirectoryClient();
+        client.runArgs(args);
+    }
+
+    /**
+     * Run with arguments.
+     *
+     * @param args arguments.
+     */
+    public void runArgs(String[] args) {
+        CommandLineParser clParser = new DefaultParser();
+        CommandLine line;
+        String host;
+        String port;
+        CMD cmd;
+        String[] cmdArgs;
+        try {
+            // parse the command line arguments
+            line = clParser.parse(options, args);
+            if (line.hasOption("help")) {
+                printUsage();
+            }
+            if (line.hasOption("server")) {
+                String server = line.getOptionValue("server");
+                try {
+                    host = server.split(":")[0];
+                    port = server.split(":")[1];
+                    System.out.printf("server %s \n", server);
+                }catch (Exception e){
+                    throw new ParseException(String.format("wrong -server option %S\n",server));
+                }
+                ServiceDirectory.getServiceDirectoryConfig().setProperty(
+                        DirectoryServiceRestfulClient.SD_API_SD_SERVER_FQDN_PROPERTY,
+                        host);
+                ServiceDirectory.getServiceDirectoryConfig().setProperty(
+                        DirectoryServiceRestfulClient.SD_API_SD_SERVER_PORT_PROPERTY,
+                        port);
+            }
+            if (line.hasOption("exec")) {
+                String commandString = line.getOptionValue("exec");
+                try {
+                    cmdArgs = parseCommand(commandString);
+                    cmd = CMD.valueOf(cmdArgs[0]);
+                    if (cmd.sizeOfArgs + 1 != cmdArgs.length) {
+                        throw new ParseException(
+                                String.format("Wrong arguments number for command %s, required %s but %s \n",
+                                        cmdArgs[0], cmd.sizeOfArgs, cmdArgs.length - 1));
+                    }
+                    switch (cmd) {
+                        case getInstanceOf:
```

Comment:
yes, be intend to mark the getAllServices is not supported when testing . clean it up it today.